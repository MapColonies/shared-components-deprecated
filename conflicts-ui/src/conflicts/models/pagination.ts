import { types, Instance } from 'mobx-state-tree';

export const pagination = types
  .model({
    page: types.optional(types.number, 0),
    itemsPerPage: types.optional(types.number, 5),
    totalItemsCount: types.optional(types.number, 0),
  })
  .actions((self) => ({
    reset () {
      self.page = 0;
    },
    setPage (page: number) {
      self.page = page;
    },
    setItemsPerPage (itemsPerPage: number) {
      self.itemsPerPage = itemsPerPage;
      self.page = 0;
    },
    setTotalItems (count: number) {
      self.totalItemsCount = count;
    },
  }));

export interface IPaginationStore extends Instance<typeof pagination> {}
