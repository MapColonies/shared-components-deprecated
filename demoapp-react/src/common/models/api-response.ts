export interface ApiHttpError {
  statusCode: number;

  message: string;
}

export interface ApiHttpResponse<T = {}> {
  success: boolean;

  data: T;

  error: ApiHttpError | {};
}
