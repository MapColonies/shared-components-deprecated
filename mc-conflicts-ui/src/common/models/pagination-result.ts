export interface PaginationResult<T> {
  currentPage: number;

  perPage: number;

  offset: number;

  total: number;

  lastPage?: number;

  data: T

  from?: number;

  to?: number;
}