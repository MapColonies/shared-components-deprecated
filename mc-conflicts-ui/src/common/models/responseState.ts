export enum ResponseState {
  "pending" = "pending",
  "done" = "done",
  "error" = "error",
}
