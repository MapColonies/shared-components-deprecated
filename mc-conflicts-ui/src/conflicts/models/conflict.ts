import { types, Instance } from "mobx-state-tree";
import { Geometry } from "geojson";

export const Conflict = types.model({
  id: types.identifier,
  source_server: types.string,
  target_server: types.string,
  source_entity: types.frozen(),
  target_entity: types.frozen(),
  description: types.string,
  location: types.frozen<Geometry>(),
  has_resolved: types.boolean,
  resolved_at: types.maybeNull(types.Date),
  resolution_id: types.maybeNull(types.string),
  created_at: types.Date,
  updated_at: types.Date,
  deleted_at: types.maybeNull(types.Date),
});

export interface IConflict extends Instance<typeof Conflict> {}
