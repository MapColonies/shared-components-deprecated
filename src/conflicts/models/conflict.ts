export interface Conflict {
  id: string;

  source_server: string;

  target_server: string;

  source_entity: object;

  target_entity: object;

  description: string;

  location?: Location;

  has_resolved: boolean;

  resolved_at?: Date;

  resolution_id?: string;

  created_at: Date;

  updated_at: Date;

  deleted_at?: Date;
}