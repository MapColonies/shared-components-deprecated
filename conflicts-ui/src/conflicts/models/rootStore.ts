import { types, Instance, getEnv, onAction } from 'mobx-state-tree';
import { useContext, createContext } from 'react';
import { ResponseState } from '../../common/models/ResponseState';
import { ConflictStore, conflictResponse } from './conflictStore';

type fetchConflicts = (url:string, params:object) => Promise<conflictResponse>;

export const baseRootStore = types
  .model({
    conflictsStore: types.optional(ConflictStore, {
      state: ResponseState.PENDING,
      searchParams: {},
    }),
    // mapStore: types.optional(ConflictMapState, {})
  })
  .views((self) => ({
    get fetch (): fetchConflicts {
      return getEnv(self).fetch as fetchConflicts;
    },
  }));

export const rootStore = baseRootStore.actions((self) => ({
  afterCreate (): void {
    self.conflictsStore.fetchConflicts();

    onAction(
      self,
      (call) => {
        if (call.name === 'setItemsPerPage' || call.name === 'setPage') {
          self.conflictsStore.fetchConflicts();
        }
      },
      true
    );
  },
}));
export interface IBaseRootStore extends Instance<typeof baseRootStore> {}
export interface IRootStore extends Instance<typeof rootStore> {}
const RootStoreContext = createContext<null | IRootStore | IBaseRootStore>(
  null
);

export const StoreProvider = RootStoreContext.Provider;
export const useStore = (): IRootStore | IBaseRootStore => {
  const store = useContext(RootStoreContext);
  if (store === null) {
    throw new Error('Store cannot be null, please add a context provider');
  }
  return store;
};
