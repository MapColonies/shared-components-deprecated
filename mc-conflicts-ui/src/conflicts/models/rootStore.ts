import { types, Instance, getEnv, onAction } from "mobx-state-tree";
import { useContext, createContext } from 'react';
import { ConflictStore } from './conflictStore';
import { ResponseState } from "../../common/models/ResponseState";

export const baseRootStore = types
  .model({
    conflictsStore: types.optional(ConflictStore, { state: ResponseState.PENDING, searchParams: {} }),
    // mapStore: types.optional(ConflictMapState, {})
  })
  .views(self => ({
    get fetch() {
      return getEnv(self).fetch;
    }
  }));

export const rootStore = baseRootStore.actions(self => ({
  afterCreate() {
    self.conflictsStore.fetchConflicts();
    
      onAction(self, call => {
        if (call.name === 'setItemsPerPage' || call.name === 'setPage'){
          self.conflictsStore.fetchConflicts();
        }
      }, true)
  }
}));
export interface IBaseRootStore extends Instance<typeof baseRootStore> { };
export interface IRootStore extends Instance<typeof rootStore> { };
const RootStoreContext = createContext<null | IRootStore | IBaseRootStore>(null);

export const StoreProvider = RootStoreContext.Provider;
export const useStore = () => {
  const store = useContext(RootStoreContext);
  if (store === null) {
    throw new Error("Store cannot be null, please add a context provider");
  }
  return store;
}