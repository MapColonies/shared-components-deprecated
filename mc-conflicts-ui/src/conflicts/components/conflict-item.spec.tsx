import React from 'react'
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import ConflictItem from './conflict-item'
import { IConflict } from '../models/conflict';

const conflict: IConflict = {
  id: "7e73d05b-8159-4e51-a302-78aa6b8fbd97",
  source_server: "server1",
  target_server: "server2",
  source_entity: {},
  target_entity: {},
  description: "some desc 1",
  has_resolved: false,
  resolved_at: null,
  location: { type: "Point", "coordinates": [35.2137, 31.7683] },
  resolution_id: null,
  created_at: new Date(1234567890),
  updated_at: new Date(1234567891),
  deleted_at: null
}

it('renders correctly', () => {
  // const store = rootStore.create({}, { fetch: conflictFetcher })
  const tree = renderer
    .create(<ConflictItem conflict={conflict} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});