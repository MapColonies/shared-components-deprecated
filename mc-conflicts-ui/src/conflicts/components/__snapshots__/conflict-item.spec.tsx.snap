// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1"
  >
    #7e73d05b-8159-4e51-a302-78aa6b8fbd97
  </h6>
  <div>
    server1
  </div>
  <div>
    server2
  </div>
  <div>
    some desc 1
  </div>
</div>
`;
